File: cellab_devkit-1.0 (1)/rules_java/bob.java
/*

        This is modelled on the Hodgepodge rule of Gerhardt and
        Schuster, but is not a close enough model to produce a
        Zhabotinsky reaction except after extremely long run times.

        The start pattern used is the Shroud of Turing visage of
        "Bob".  Bob is the High Epopt of the Church of the SubGenius.
        For more information about Bob and the Church, send $1 and a
        long stamped self-addressed envelope to:

                The SubGenius Foundation
                Box 140306
                Dallas, TX 75214

        The image of Bob is a registered trademark of the Church of
        the Subgenius and is used by special arrangement with Douglas
        St. Claire Smith, a.k.a. Ivan Stang.  Inquiries about
        further usage of Bob's image should be directed to Mr. Smith
        c/o The SubGenius Foundation.

*/

class Bob extends ruletable {

    void jcruleModes() {
        setPatternRequest("bob");
        setPaletteRequest("bob");
    }

    int jcrule(int oldstate) {
        int EightSum = nw + n + ne + e + se + s + sw + w,
            Sickness, NewState;

        if (oldstate == 0) {
            if (EightSum == 0) {
                NewState = 0;
            } else {
                NewState = EightSum | 1;
            }
        } else {
            Sickness = oldstate >> 1;
            if (Sickness == 64) {
                NewState = 0;
            } else {
                Sickness = Sickness + EightSum + 3;
                if (Sickness > 64) {
                    Sickness = 64;
                }
                NewState = (Sickness << 1) | 1;
            }
        }
        return NewState;
    }
}

public class bob {
    public static void main(String args[]) {
        (new Bob()).generateRuleFile("bob");
    }
}

File: references/cellab_devkit-1.0 (1)/ruledefs/bob.js
/*

        This is modelled on the Hodgepodge rule of Gerhardt and
        Schuster, but is not a close enough model to produce a
        Zhabotinsky reaction except after extremely long run times.

        The start pattern used is the Shroud of Turing visage of
        "Bob".  Bob is the High Epopt of the Church of the SubGenius.
        For more information about Bob and the Church, send $1 and a
        long stamped self-addressed envelope to:

                The SubGenius Foundation
                Box 181417,
		Cleveland Heights, OH  44118-1417
                USA

        The image of Bob is a registered trademark of the Church of
        the Subgenius and is used by special arrangement with Douglas
        St. Claire Smith, a.k.a. Ivan Stang.  Inquiries about
        further usage of Bob's image should be directed to Mr. Smith
        c/o The SubGenius Foundation.

*/

    rule.worldtype = 1;          // 2D torus world
    rule.ruleName = "bob";
    rule.patreq = "bob";
    rule.palreq = "bob";
    
    function bob(oldstate,     nw, n  , ne,
                               w, self, e,
                               sw, s  , se) {

        var eightSum = nw + n + ne + e + se + s + sw + w,
            sickness, newState;

        if (oldstate == 0) {
            if (eightSum == 0) {
                newState = 0;
            } else {
                newState = eightSum | 1;
            }
        } else {
            sickness = oldstate >> 1;
            if (sickness == 64) {
                newState = 0;
            } else {
                sickness = sickness + eightSum + 3;
                if (sickness > 64) {
                    sickness = 64;
                }
                newState = (sickness << 1) | 1;
            }
        }
        return newState;
    }
